phpBB/includes/acp/acp_storage.php

```diff
@@ -315,4 +348,42 @@ protected function update_storage_config($storage_name)
 			$this->config->set('storage\\' . $storage_name . '\\config\\' . $definition, $this->get_new_definition($storage_name, $definition));
 		}
 	}
+
+	protected function get_current_adapter($storage_name)
+	{
+		$provider = $this->get_current_provider($storage_name);
+		$provider_class = $this->provider_collection->get_by_class($provider);
+
+		$adapter = $this->adapter_collection->get_by_class($provider_class->get_adapter_class());
+		$definitions = $this->get_provider_options($provider);
+
+		$options = [];
+		foreach (array_keys($definitions) as $definition)
+		{
+			$options[$definition] = $this->get_current_definition($storage_name, $definition);
+		}
+
+		$adapter->configure($options);
```

Comment:
Probably i will need to add here in the future
$adapter->set_storage($storage_name);